### ISSUE TITLE
Too slow when output multiple lines

### ISSUE THREAD

**Issue by tankgit** [Eugeny/tabby#6394]
When the shell outputs somthing with multiple lines, Tabby seems to show the output line by line rather than giving all lines instantly, even `ls -l` command could take secs to show hundreds of files.

---

PS: new version of 1.0.177 is involved with the problem, rollback to 1.0.176 solve the problem.

**Comment by W0n9** [comment]
I faced the same problem and it severely affected the experience when using some CUI program such as `htop`.

**Comment by Eugeny** [comment]
Could you please record a performance trace by opening Devtools from Settings -> Application -> Devtools and then following https://developer.chrome.com/docs/devtools/evaluate-performance/reference/ ?

**Comment by W0n9** [comment]
Here is my performance trace file.  
[Profile-20220514T233104.json](https://gist.github.com/W0n9/3abc1972a88ef392dce51db206842dd1)

**Comment by zheng2010** [comment]
我也遇到了同样的问题

**Comment by zheng2010** [comment]
我也遇到了同样的问题哦

**Comment by alexio777** [comment]
I have the same problem

**Comment by ywang-wnlo** [comment]
the same as above

**Comment by khaledh** [comment]
Same here. I had to revert to 176 until this gets fixed.

**Comment by jxtyq69** [comment]
it is fixed in 1.0.178 , thank you for developer

**Comment by wudisheng** [comment]
I encountered the same problem with .179 on Windows 11 x64, sometimes the symptom is much worse ---

- Tabby enters a status not responding to user inputs for the exact tab (while still printing lots of lines quickly overflow the scroll buffer); AND
- Tabby starts to use a significant amount of CPU (on a very high-end PC); AND
- Even if I repeatedly enter something like Ctrl-C immediately after entering the command that triggers a lot of output, it takes **10 seconds** or **20 seconds** or even longer to stop the output (especially when the output lines are **colored**).

I tried rolling back to .176 and it seems better (still in verification).

A typical command to trigger the issue `fgrep . -Inr / --color`.

I can grab and attach a performance trace later (after I have time to reinstall .179 debug it) if it sounds necessary to you.
